### PR TITLE
*: support 'admin show ddl jobs <number>' grammar

### PR DESCRIPTION
**Pull request by crazycs520**
## What have you changed? (mandatory)
[issue#7025](https://github.com/pingcap/tidb/issues/7025)
add "admin show ddl jobs [number]" grammar support.
this is use to support [parallel ddl test in schrodinger-test](https://github.com/pingcap/schrodinger-test/pull/72)
eg:
```sql
mysql root@127.0.0.1:(none)> admin show ddl jobs
+--------+---------+--------------------------------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
| JOB_ID | DB_NAME | TABLE_NAME                           | JOB_TYPE     | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | START_TIME                        | STATE  |
+--------+---------+--------------------------------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
| 545    | test    | d5ea4e33-4fe2-46ae-accb-82610ea1e0b1 | add index    | public       | 1         | 489      | 0         | 2018-07-10 20:07:04.144 +0800 CST | synced |
| 544    | test    | a9ce39c0-f856-4ac0-9c80-78e32ac44481 | create table | public       | 1         | 543      | 0         | 2018-07-10 20:07:04.067 +0800 CST | synced |
| 542    | test    | 0284e4b1-56e9-43c1-bcf5-0a82126fbaee | drop column  | none         | 1         | 517      | 0         | 2018-07-10 20:07:03.366 +0800 CST | synced |
| 541    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add column   | public       | 1         | 474      | 0         | 2018-07-10 20:07:03.195 +0800 CST | synced |
| 540    | test    | 0284e4b1-56e9-43c1-bcf5-0a82126fbaee | drop index   | none         | 1         | 517      | 0         | 2018-07-10 20:07:03.139 +0800 CST | synced |
| 539    | test    | afe54691-648f-4574-a05a-f706b368e07b | add index    | public       | 1         | 496      | 1         | 2018-07-10 20:07:02.621 +0800 CST | synced |
| 538    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add column   | public       | 1         | 474      | 0         | 2018-07-10 20:07:02.506 +0800 CST | synced |
| 537    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add column   | public       | 1         | 474      | 0         | 2018-07-10 20:07:02.032 +0800 CST | synced |
| 536    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add index    | public       | 1         | 474      | 0         | 2018-07-10 20:07:01.86 +0800 CST  | synced |
| 535    | test    | d5ea4e33-4fe2-46ae-accb-82610ea1e0b1 | drop index   | none         | 1         | 489      | 0         | 2018-07-10 20:07:01.694 +0800 CST | synced |
+--------+---------+--------------------------------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
10 rows in set
Time: 0.079s
mysql root@127.0.0.1:(none)> admin show ddl jobs 20
+--------+---------+--------------------------------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
| JOB_ID | DB_NAME | TABLE_NAME                           | JOB_TYPE     | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | START_TIME                        | STATE  |
+--------+---------+--------------------------------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
| 545    | test    | d5ea4e33-4fe2-46ae-accb-82610ea1e0b1 | add index    | public       | 1         | 489      | 0         | 2018-07-10 20:07:04.144 +0800 CST | synced |
| 544    | test    | a9ce39c0-f856-4ac0-9c80-78e32ac44481 | create table | public       | 1         | 543      | 0         | 2018-07-10 20:07:04.067 +0800 CST | synced |
| 542    | test    | 0284e4b1-56e9-43c1-bcf5-0a82126fbaee | drop column  | none         | 1         | 517      | 0         | 2018-07-10 20:07:03.366 +0800 CST | synced |
| 541    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add column   | public       | 1         | 474      | 0         | 2018-07-10 20:07:03.195 +0800 CST | synced |
| 540    | test    | 0284e4b1-56e9-43c1-bcf5-0a82126fbaee | drop index   | none         | 1         | 517      | 0         | 2018-07-10 20:07:03.139 +0800 CST | synced |
| 539    | test    | afe54691-648f-4574-a05a-f706b368e07b | add index    | public       | 1         | 496      | 1         | 2018-07-10 20:07:02.621 +0800 CST | synced |
| 538    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add column   | public       | 1         | 474      | 0         | 2018-07-10 20:07:02.506 +0800 CST | synced |
| 537    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add column   | public       | 1         | 474      | 0         | 2018-07-10 20:07:02.032 +0800 CST | synced |
| 536    | test    | f94383a9-bd1c-4a1e-bd1f-ec44af3e3c1e | add index    | public       | 1         | 474      | 0         | 2018-07-10 20:07:01.86 +0800 CST  | synced |
| 535    | test    | d5ea4e33-4fe2-46ae-accb-82610ea1e0b1 | drop index   | none         | 1         | 489      | 0         | 2018-07-10 20:07:01.694 +0800 CST | synced |
| 534    | test    |                                      | drop table   | none         | 1         | 502      | 0         | 2018-07-10 20:07:01.673 +0800 CST | synced |
| 533    | test    | d5ea4e33-4fe2-46ae-accb-82610ea1e0b1 | add index    | public       | 1         | 489      | 3         | 2018-07-10 20:07:01.312 +0800 CST | synced |
| 532    | test    | afe54691-648f-4574-a05a-f706b368e07b | add column   | public       | 1         | 496      | 0         | 2018-07-10 20:07:01.311 +0800 CST | synced |
| 531    | test    | d5ea4e33-4fe2-46ae-accb-82610ea1e0b1 | drop column  | none         | 1         | 489      | 0         | 2018-07-10 20:07:00.873 +0800 CST | synced |
| 530    | test    | afe54691-648f-4574-a05a-f706b368e07b | add column   | public       | 1         | 496      | 0         | 2018-07-10 20:07:00.872 +0800 CST | synced |
| 529    | test    |                                      | add index    | public       | 1         | 502      | 1         | 2018-07-10 20:07:00.419 +0800 CST | synced |
| 528    | test    | 0284e4b1-56e9-43c1-bcf5-0a82126fbaee | add column   | public       | 1         | 517      | 0         | 2018-07-10 20:07:00.417 +0800 CST | synced |
| 527    | test    | 0284e4b1-56e9-43c1-bcf5-0a82126fbaee | add column   | public       | 1         | 517      | 0         | 2018-07-10 20:06:59.972 +0800 CST | synced |
| 526    | test    | 0284e4b1-56e9-43c1-bcf5-0a82126fbaee | add index    | public       | 1         | 517      | 0         | 2018-07-10 20:06:59.97 +0800 CST  | synced |
| 525    | test    |                                      | add column   | public       | 1         | 502      | 0         | 2018-07-10 20:06:59.719 +0800 CST | synced |
+--------+---------+--------------------------------------+--------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
20 rows in set
Time: 0.064s
mysql root@127.0.0.1:(none)>
```

## What are the type of the changes (mandatory)?

- New feature (non-breaking change which adds functionality)

## How has this PR been tested (mandatory)?
unit test

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -247,7 +248,7 @@ func (e *ShowDDLJobQueriesExec) Open(ctx context.Context) error {
 		return errors.Trace(err)
 	}
 	// TODO: need to return the job that the user needs.
-	historyJobs, err := admin.GetHistoryDDLJobs(e.ctx.Txn())
+	historyJobs, err := admin.GetHistoryDDLJobs(e.ctx.Txn(), 10)
```
Comment (zimulala) [inline executor/executor.go]: Could we define `10` as a constant?

File: executor/executor_test.go [hunk 2]
```diff
@@ -181,13 +181,23 @@ func (s *testSuite) TestAdmin(c *C) {
...
 	c.Assert(row.GetInt64(0), Equals, historyJobs[0].ID)
 	c.Assert(err, IsNil)
 
+	r, err = tk.Exec("admin show ddl jobs 20")
+	c.Assert(err, IsNil)
+	chk = r.NewChunk()
+	err = r.Next(ctx, chk)
+	c.Assert(err, IsNil)
+	row = chk.GetRow(0)
+	c.Assert(row.Len(), Equals, 10)
```
Comment (zimulala) [inline executor/executor_test.go]: Why is its value 10 is not 20?

Comment (jackysp) [reply executor/executor_test.go]: I think it the first-row length.

Comment (crazycs520) [reply executor/executor_test.go]: ye~

Comment (zimulala) [reply executor/executor_test.go]: Do we need to check the number of jobs in the result?

File: util/admin/admin.go [hunk 3]
```diff
@@ -137,18 +137,21 @@ func GetDDLJobs(txn kv.Transaction) ([]*model.Job, error) {
 // MaxHistoryJobs is exported for testing.
 const MaxHistoryJobs = 10
 
+// DefHistoryJobs is default value of the default number of history job
+const DefHistoryJobs = 10
```
Comment (XuHuaiyu) [inline util/admin/admin.go]: s/DefHistoryJobs/DefHistoryJobsNum?

File: executor/executor.go [hunk 4]
```diff
@@ -288,8 +289,10 @@ func (e *ShowDDLJobsExec) Open(ctx context.Context) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-
-	historyJobs, err := admin.GetHistoryDDLJobs(e.ctx.Txn())
+	if e.jobNumber == 0 {
+		e.jobNumber = 10
```
Comment (XuHuaiyu) [inline executor/executor.go]: s/ 10/ admin/DefHistoryJobs?

File: parser/parser_test.go [hunk 5]
```diff
@@ -413,6 +413,7 @@ func (s *testParserSuite) TestDMLStmt(c *C) {
 		// for admin
 		{"admin show ddl;", true},
 		{"admin show ddl jobs;", true},
+		{"admin show ddl jobs 20;", true},
```
Comment (XuHuaiyu) [inline parser/parser_test.go]: add test for
`admin show ddl jobs -1;`

File: executor/executor_test.go (hunk 2 above)
Comment (crazycs520) [reply executor/executor_test.go]: The number of jobs in the result cannot be ensured, it is related to the actual DDL jobs.

File: util/admin/admin.go (hunk 3 above)
Comment (crazycs520) [reply util/admin/admin.go]: done

File: parser/parser_test.go (hunk 5 above)
Comment (crazycs520) [reply parser/parser_test.go]: done

File: executor/executor.go [hunk 6]
```diff
@@ -247,7 +248,7 @@ func (e *ShowDDLJobQueriesExec) Open(ctx context.Context) error {
 		return errors.Trace(err)
 	}
 	// TODO: need to return the job that the user needs.
```
Comment (zz-jason) [inline executor/executor.go]: this todo can be removed?

File: util/admin/admin.go [hunk 7]
```diff
@@ -137,18 +137,21 @@ func GetDDLJobs(txn kv.Transaction) ([]*model.Job, error) {
 // MaxHistoryJobs is exported for testing.
 const MaxHistoryJobs = 10
 
+// DefHistoryJobsNum is default value of the default number of history job
+const DefHistoryJobsNum = 10
```
Comment (zz-jason) [inline util/admin/admin.go]: how about s/DefHistoryJobsNum/DefNumHistoryJobs/ ?

File: util/admin/admin.go [hunk 8]
```diff
@@ -137,18 +137,21 @@ func GetDDLJobs(txn kv.Transaction) ([]*model.Job, error) {
...
+
 // GetHistoryDDLJobs returns the DDL history jobs and an error.
-// The maximum count of history jobs is MaxHistoryJobs.
-func GetHistoryDDLJobs(txn kv.Transaction) ([]*model.Job, error) {
+// The maximum count of history jobs is num.
+func GetHistoryDDLJobs(txn kv.Transaction, num int) ([]*model.Job, error) {
```
Comment (zz-jason) [inline util/admin/admin.go]: how about s/num/maxNumJobs/ ?

Comment (crazycs520) [reply util/admin/admin.go]: done

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@zimulala @jackysp @XuHuaiyu PTAL

**Pull request merged**